solar_system_project/solar_system.rb

```diff
@@ -0,0 +1,68 @@
+
+class SolarSystem
+  attr_reader :star_name, :planets
+
+  def initialize(star_name)
+    @star_name = star_name
+    @planets = []
+  end
+
+  def add_planet(planet)
+    planets.push(planet)
+  end
+
+  def list_planets
+    number = 1
+    planet_list = "Planets orbiting #{star_name}:\n"
+    planets.each do |x|
+      planet_list += "#{number}: #{x.name}\n"
+      number += 1
+    end
+    return planet_list
+  end
+
+  def find_planet_by_name(planet_string)
+    formatted_planet_string = planet_string.split(/ |\_/).map(&:capitalize).join(" ")
+    planets.each do |x|
+      if x.name == formatted_planet_string
+        return x
+      end
+    end
+  end
+
+  def planet_details
+    puts "Which planet would you like to learn more about?: "
+    planet_choice = gets.chomp
+    return find_planet_by_name(planet_choice).summary
+  end
+
+  def distance_between(planet1, planet2)
+    return (planet1.distance_from_sun_km.to_i - planet2.distance_from_sun_km.to_i).abs
+  end
+
+  def distance_between_planets
+    puts "What is the first planet you would like to compare?: "
+    planet1 = gets.chomp.split(/ |\_/).map(&:capitalize).join(" ")
+    puts "What is the second planet you would like to compare?: "
+    planet2 = gets.chomp.split(/ |\_/).map(&:capitalize).join(" ")
+    return "There are #{distance_between(find_planet_by_name(planet1), find_planet_by_name(planet2))}km between #{planet1} and #{planet2}."
+  end
+
+  def add_user_planet
+    puts "What is the name of the planet?: "
```

Comment:
This sort of data gathering and user interaction usually gets put with the rest of the command line interface, as it makes it easier to see what data manipulation SolarSystem is doing.